### ISSUE TITLE
Misreported cycle number in `gradients` file

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [grimme-lab/xtb#889]
**Describe the bug**
The `gradient` file generated with `--grad` always reports that the results are for cycle 1 even when they are (presumably) for a different cycle.

The top of the `gradient` file in the example below is as follows:

```
$grad
  cycle =      1    SCF energy =   -26.43824279253   |dE/dxyz| =  0.000389
```

Note how it says cycle 1. The SCF energy, based on the log report below, seems to be for the final calculation in the set of cycles and not the first.

**To Reproduce**
`xtb coord.xyz --opt --grad > output.log`

Output files below:

[samples.zip](https://github.com/grimme-lab/xtb/files/13193735/samples.zip)

**Expected behaviour**
The correct cycle number should be reported.

**Additional context**
None.

**Comment by awvwgk** [comment]
The cycle reported here is a dummy rather than the actual cycle number.

**Comment by Andrew-S-Rosen** [comment]
Yup. I just thought it could be potentially confusing to the user.
